### PR TITLE
Add forms 

### PR DESCRIPTION
**Pull request by Zel-hub7**
1. I have created forms for posting.
2. I have created forms for comments.
3. I created a button of like

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @Zel-hub7 🙋‍♀️,  

While you made a great effort in this project, I can't review your code.

## Invalid Code Review Request
 
You have submitted a project that has a README that is not fully documented. Here is a key point to consider when documenting your README.

- Under the `Getting Started` section, kindly add commands for how to setup the database and run the project locally. The `Usage` section should include commands like `rails db:setup / rails db:create & rails db:migrate` and `rails s` for running the project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -11,9 +13,30 @@ def show
...
+  def create
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      flash.now[:error] = @post.errors.full_messages.join(', ')
+      render :new
+    end
+  end
```
Comment (Whoistolu) [inline app/controllers/posts_controller.rb:31]: - [ ] [OPTIONAL] Great job using the `.build` method which is an ActiveRecord method. However, it doesn't persist the object to the database. Using @user.posts.build will build a new post associated with the @user object, but it won't save it to the database until you explicitly call save.

You can just use the `.new`:
``` ruby
 def new
      @post = Post.new
    end

    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```
- This pattern is perfectly valid and widely used in Rails applications. The new action is responsible for rendering the form to create a new post, and the create action is responsible for processing the form submission and saving the new post to the database.

- Also, If you want to both build and save the object in one step, you can use create instead. Using create is a shorthand for build followed by save. It creates a new object, assigns the attributes, and then immediately saves it to the database.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    @user = User.find(params[:user_id])
+  end
+
+  def set_post
+    @post = Post.find(params[:post_id])
+  end
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
```
Comment (Whoistolu) [inline app/controllers/comments_controller.rb]: - [ ] [REQUIRED] Great effort adding the `create` method to the comments controller. However, you need to add the `new` method as well. They will be handling the creation of a new comments and its subsequent persistence. It uses the `new` action to instantiate a new Comment object and the create action to handle the creation and saving of a new comment associated with the @user.

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   root 'users#index'
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :comments, only: [:create]
```
Comment (Whoistolu) [inline config/routes.rb]: - [ ] [REQUIRED] After you have added the `new` method to the comments controller, you will also need to add it to the routes just as you have done for the Post. Let's fix this  well.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Zel-hub7 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### highlight
🔆 ApplicationController contains `#current_user` method
🔆 A post is created on behalf of the currently logged in user (current_user) of the app
🔆 There are pair of routes in the config/routes.rb file for creating posts
🔆 The likes controller have a create actions and a route in the config/routes.rb file for creating like

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (peterrobert) [reply app/controllers/comments_controller.rb]: Great work adding the new method.

File: config/routes.rb (hunk 3 above)
Comment (peterrobert) [reply config/routes.rb]: Great work adding the new comment route

**Review by peterrobert — APPROVED** [review]
Hi @Zel-hub7, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️

[ ] Great work making the changes requested by the previous reviewer.


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**